### DISCUSSION TITLE
How do I configure autocreate? where? Is it documented?

### DISCUSSION THREAD

**Discussion by w2n1ck** [DependencyTrack/dependency-track#870] [Q&A]
you have two options:
- specify the uuid of project (will not auto create obviously)
- specify the name and version of the project (it will automatically be created if it doesn't exist)

## Sample pipeline
```groovy
node {
    stage "Publish Dependency-Track results"
    
    dependencyTrackPublisher artifact: 'bom.xml', artifactType: 'bom', projectName: 'Acme Example', projectVersion: '1.0', synchronous: true
    
    //dependencyTrackPublisher artifact: 'bom.xml', artifactType: 'bom', projectId: '3a38aedf-e9e9-4e0a-8913-2d99951aa76d', synchronous: true
}
```

## Console output
```
Entering stage Publish Dependency-Track results
Proceeding
[Pipeline] dependencyTrackPublisher
[DependencyTrack] Publishing artifact to Dependency-Track - http://192.168.1.16:8080
[DependencyTrack] The artifact was successfully published
[DependencyTrack] Polling Dependency-Track for BOM processing status
[DependencyTrack] Looking up id of newly created project
[DependencyTrack] Processing findings
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
```

The projectName and projectVersion can obviously be variables injected from your build as well.

_Originally posted by @stevespringett in https://github.com/DependencyTrack/dependency-track/issues/499#issuecomment-552738613_

**Comment by stevespringett** [comment]
Its documented in the swagger.

https://docs.dependencytrack.org/integrations/rest-api/

Also note, the Jenkins plugin is not maintained here. It has its own maintainer independent of this project.

**Comment by w2n1ck** [comment]
No, I'm using the DT Jenkins Plugin, and I want the plugin to create the project automatically. Of course, give to PROJECT_CREATION_UPLOAD.

**Comment by stevespringett** [comment]
Refer to https://github.com/jenkinsci/dependency-track-plugin

**Comment by DrakezulsMinimalism** [comment]
POSTing to the REST API use "autoCreate=true" and specify "projectName=YOUR_NEW_NAME_HERE". Optionally you can also specify a "projectVersion".

I tested the following (DependencyTrack 4.2.1):
> curl -X "POST" "<backend-url>/api/v1/bom" \\
> &nbsp;&nbsp;&nbsp;&nbsp;-H "Content-Type: multipart/form-data; charset=utf-8; boundary=\_\_X_CURL_BOUNDARY\_\_" \\
> &nbsp;&nbsp;&nbsp;&nbsp;-H "X-API-Key: **%your_api_key_with_permission_PROJECT_CREATION_UPLOAD%**" \\
> &nbsp;&nbsp;&nbsp;&nbsp;-F "projectName=**%your project name%**" \\
> &nbsp;&nbsp;&nbsp;&nbsp;-F "**autoCreate=true**" 
> &nbsp;&nbsp;&nbsp;&nbsp;-F "bom=@bom.xml"
